Assets/Scripts/Player/PlayerShootingBehaviour.cs

```diff
@@ -10,14 +10,47 @@ public class PlayerShootingBehaviour : MonoBehaviour,IShootBullet
 
     public PlayerTankView playerRef;
 
+    private int  bulletCount;
     private void Start()
     {
         playerRef = PlayerTankSpawner.Instance.ReturnView();
     }
     public  void ShootBullet()
     {
-        if(playerRef!=null)
-        bulletSpawner.SpawnBullet(spawnPoint,BulletEnum.PlayerBullet,this.gameObject);
-      
+
+        if (playerRef != null)
+        {
+            bulletSpawner.SpawnBullet(spawnPoint, BulletEnum.PlayerBullet, this.gameObject);
+
+            bulletCount++;
+            
+            
+        }
+        if (bulletCount >= 10)
+        {
+            CheckBulletAchivementUnlock(bulletCount);
+        }
+
+    }
+
+
+    public void CheckBulletAchivementUnlock(int bullets)
```

Comment:
you can create a generic method which checks what type of achievement is invoked and then properly indicates the message. Since you have created enums , you can use them to identify the type of achievement happened and respectively invoke the achievement message.